backend/location/utility.py

```diff
@@ -0,0 +1,209 @@
+            switched_point = (point[1], point[0])
+            switched_ring.append(switched_point)
+        switched_poly.append(LinearRing(switched_ring))
+
+    return Polygon(*switched_poly)
+
+
+# Commenter: Chris
+# The code for formatting a location is the same code as in the frontend pythonized.
+# The reason why we have it here as well is to format locations that we retrieved
+# with direct requests from the backend to nominatim.
+# This means changes made here or in the equivelant function in the frontend also need to 
+# be applied to the other function so they stay consistent
+# This whole setup is less than ideal and should be changed in the future.
+
+def format_location(location_string):
+    location_object = json.loads(location_string)[0]
+    location_name = format_location_name(location_object)
+    return {
+        'type': location_object['geojson']['type'],
+        'place_id': location_object['place_id'],
+        'osm_id': location_object['osm_id'],
+        'name': location_name['name'],
+        'city': location_name['city'],
+        'state': location_name['state'],
+        'country': location_name['country'],
+        'geojson': location_object['geojson'],
+        'coordinates': location_object['geojson']['coordinates']
+    }
+
+def format_location_name(location):
+    first_part_order = [
+        "village",
+        "town",
+        "city_district",
+        "district",
+        "suburb",
+        "borough",
+        "subdivision",
+        "neighbourhood",
+        "place",
+        "city",
+        "municipality",
+        "county",
+        "state_district",
+        "province",
+        "state",
+        "region"
+    ]
+    middle_part_order = [
+        "city_district",
+        "district",
+        "suburb",
+        "borough",
+        "subdivision",
+        "neighbourhood",
+        "town",
+        "village"
+    ]
+    if is_country(location):
+        return {
+            'country': location['address']['country'],
+            'city': "",
+            'state': "",
+            'name': location['display_name']
+        }
+    
+    middle_part_suffixes = ["city", "state"]
+    first_part = get_first_part(location['address'], first_part_order)
+    middle_part = get_middle_part(location['address'], middle_part_order, middle_part_suffixes)
+    return {
+        'city': first_part,
+        'state': middle_part,
+        'country': location['address']['country'],
+        'name': first_part + ", " + middle_part + (", " if len(middle_part) > 0 else "") + location['address']['country'],
+    }
+
+def is_country(location):
+    if location['type'] != "administrative":
+        return False
+    # short circuit if the address contains any information other than country and country code
+    for key in location['address'].keys():
+        if key not in ["country", "country_code"]:
+            return False       
+    return True
+
+def get_first_part(address, order):
+    for el in order:
```

Comment:
Can you have object name a bit more descriptive? `el` does not make sense what it means.